core/arch/arm/sm/sm.c

```diff
@@ -26,13 +27,17 @@ bool sm_from_nsec(struct sm_ctx *ctx)
 	COMPILE_TIME_ASSERT(!(offsetof(struct sm_ctx, nsec.r0) % 8));
 	COMPILE_TIME_ASSERT(!(sizeof(struct sm_ctx) % 8));
 
-	if (!sm_platform_handler(ctx))
-		return false;
+#ifdef CFG_SM_PLATFORM_HANDLER
+	if (sm_platform_handler(ctx) == SM_HANDLER_SMC_HANDLED)
+		return SM_EXIT_TO_NON_SECURE;
+#endif
 
 #ifdef CFG_PSCI_ARM32
+	/* PSCI is the only STD service implemented */
 	if (OPTEE_SMC_OWNER_NUM(*nsec_r0) == OPTEE_SMC_OWNER_STANDARD) {
-		smc_std_handler((struct thread_smc_args *)nsec_r0, &ctx->nsec);
-		return false;	/* Return to non secure state */
+		if (smc_std_handler(thread_args, &ctx->nsec) ==
```

Comment:
Minor comment: I wouldn't mind writing this slightly more compact as:
``` C
        if (OPTEE_SMC_OWNER_NUM(*nsec_r0) == OPTEE_SMC_OWNER_STANDARD &&
            smc_std_handler(thread_args, &ctx->nsec) == SM_HANDLER_SMC_HANDLED)
                return SM_EXIT_TO_NON_SECURE;
```

Reply:
ok